js/upvotes.js

```diff
@@ -1,102 +1,92 @@
-                d('keep trying to add button');
-                if ( attachTo.find('#added_sort_btn').length > 0 )
-                {
+
+                if (attachTo.find('#added_sort_btn').length > 0) {
                     d('already attached');
                     clearInterval(add_btn_timer);
-                }
-                else
-                {
+                } else {
+                    d('adding sort_btn');
+                    var sort_btn = "<li id='added_sort_btn' class='menu_list_item'>" +
+                    "<span class='light_gray'><span><a href='#'>Sort by Votes</a></span></span></li>";
+
                     attachTo.append(sort_btn);
-                    d($('.unified_menu').size());
+                    $("#added_sort_btn").click(sort_answers);
                 }
-
-                $( "#added_sort_btn" ).click( function() {
-                    sort_questions();
-                });
             }, 200);
-        });
-        
-    }
-
-    function sort_questions () {
-        var elts = $( ".AnswerPagedList > .pagedlist_item");
-        var answers = elts
-          .not(".pagedlist_hidden")
-          .toArray();
-        var hidden_answers = $( ".AnswerPagedList > .pagedlist_hidden" );
-
-        d(Array.isArray(answers));
-
-
-            for (var i = 0; i < answers.length; ++i) {
-                answers[i].upvotes = $( answers[i] ).find( ".count" ).first().text();
-
-                if (answers[i].upvotes.includes("k"))
-                    answers[i].upvotes = parseInt(answers[i].upvotes) * 1000;
-            }
-
-
-            if (DEBUG) {
-                for (i = 0; i < answers.length; ++i) {
-                     console.log(answers[i].upvotes);
-                }
-            }
-
-            // sort the oject of answer elements by upvotes
-            answers.sort( function(a, b) {
-                if (DEBUG)
-                {
-                    console.log(typeof a.upvotes);
-                    console.log(a.upvotes);
-                }
-
-                return b.upvotes - a.upvotes;
-            });
-
-            // re render the elements based on upvote sort
-        var ajaxElem = $( ".AnswerPagedList" ).children().not(".pagedlist_item");
-            ajaxElem.detach();
-
-        d("Ajaxy thing: " + String(ajaxElem));
-
-        $( ".AnswerPagedList" ).empty();
-
-        for (i = 0; i < answers.length - 1; ++i) {
-            $( ".AnswerPagedList" ).append( answers[i] );
         }
+    });
+}
+
+// This function will sort all currently shown answers by upvotes.
+// However, if the question has a lot of answers,
+// which will only be loaded later when users scroll down,
+// then this function will need to be invoked again (somehow).
+function sort_answers() {
+    // find all shown_answers
+    var answers = $(".AnswerPagedList > .pagedlist_item");
+    var shown_answers = answers.not(".pagedlist_hidden");
+
+    // detach shown_answers from the answer list
```

Comment:
I'm a fan of the other comments as they explain some tricky bits of code, but this one seems unnecessary to me; please remove